### PR TITLE
Missing null checks added in DiffWithCustomIds method

### LINKED ISSUE #2804: Diffing_Engine: missing null check crashes DiffWithCustomIds
#### Description:

As in the title: a missing null check crashes `DiffWithCustomIds` method in particular workflows. I have identified the issue and will PR a fix right after finishing typing this issue.

### PR DESCRIPTION
**Pull request by pawelbaran**
### Issues addressed by this PR

Closes #2804

### CODE REVIEW

**Comment by pawelbaran** [comment]
@BHoMBot check compliance
@BHoMBot check null-handling

**Comment by pawelbaran** [comment]
@BHoMBot check core
@BHoMBot check installer
@BHoMBot check serialisation
@BHoMBot check versioning

File: Diffing_Engine/Compute/DiffWithCustomIds.cs [hunk 1]
```diff
@@ -263,7 +263,7 @@ private static Diff Diffing(IEnumerable<object> pastObjects, IEnumerable<string>
                         $"\nThis can also happen if the input objects come from models that were completely re-created between revisions (i.e. their IDs are completely different). In this latter case, the Diffing worked successfully but you may want to use a different ID.");
             }
 
-            if (followingObjectsIds.Intersect(pastObjectsIds).Any() && diffingConfig.ComparisonConfig.PropertiesToConsider.Any() && !modifiedObjs.Any())
+            if (followingObjectsIds.Intersect(pastObjectsIds).Any() && diffingConfig.ComparisonConfig?.PropertiesToConsider?.Any() == true && !modifiedObjs.Any())
```
Comment (alelom) [inline Diffing_Engine/Compute/DiffWithCustomIds.cs:266]:
```suggestion
            if (followingObjectsIds.Intersect(pastObjectsIds).Any() && (diffingConfig.ComparisonConfig?.PropertiesToConsider?.Any() ?? false) && !modifiedObjs.Any())
```

**Review by alelom — CHANGES_REQUESTED** [review]

**Comment by pawelbaran** [comment]
@BHoMBot check required

**Comment by pawelbaran** [comment]
Following the conversation with @alelom, the source of the issue could be solved on my side for now. On the other hand, we both agreed that nulls should be handled better in diffing, therefore another issue will be raised focusing on auditing this aspect of the code end-to-end rather than patching it up as done in here.

Following the above, I am closing both this and #2804.

**Comment by alelom** [comment]
As discussed with @pawelbaran , we decided to close this PR because not relevant to the issue.

The issue that triggered this PR is actually that some `null` values were invalidly set where they were not allowed.
To clarify this, this Issue has been raised: https://github.com/BHoM/BHoM_Engine/issues/2806. This is seen as an improvement over the Diffing (`type:feature`) rather than a bug.

**Pull request closed without merging**